src/recorder.ts

```diff
@@ -50,6 +50,14 @@ async function getBrowserInstance(options: RecorderOptions) {
   }
 }
 
+function escapeSelector(selector: string): string {
+  const escapedDblQuotedString = JSON.stringify(selector);
+  const decDblQuotes = escapedDblQuotedString.replace(/\\\"/g, '"');
+  const bumpSingleQuotes = decDblQuotes.replace(/\'/g, "\\'");
+  const trimOuterDblQuote = bumpSingleQuotes.slice(1, -1);
+  return "'" + trimOuterDblQuote + "'";
```

Comment:
It seems this still wouldn't work with quotes preceded by arbitrary escaped backslashes.

What I had in mind was simply this:

```js
function escapeSelector(selector) {
  return JSON.stringify(selector);
}
```

It seems like it'd avoid a lot of trouble — but I might be missing some context. ([`JSON.stringify` produces valid JS string literals since ES2019.](https://v8.dev/features/subsume-json))

Reply:
Yep that is what is currently there. Note that this change first does `JSON.stringify` and then tries to convert the most outer layer of `"` to `'`.
This is an attempt at prettifying the generated scripts slightly such that we do
`await click('aria/Options[role="link"]');`
instead of 
`await click("aria/Options[role=\"link\"]");`.
But maybe we should reconsider if it is worth it?

Reply:
I'm wondering the same thing — is it really worth it? `JSON.stringify()` is nice and simple, but sure, the generated code then has double quotes instead of single quotes.

As discussed offline, we could use https://github.com/mathiasbynens/jsesc which can produce JS string literals and gives fine-grained control over the output (see the `quotes` option).

Reply:
@jschfflr what do you think? I think it might be a good idea to leave the generated code slightly less pretty for now.
We could then, down the line, make the double quotes for aria properties optional on the Puppeteer side and leave them out completely (i.e. `aria/Options[role=link]`).

Reply:
I wouldn't mind to use mathias' jsesc implementation - not sure though if we have to go through the third party process if we want to include it in DevTools...
In the long run, I think it would be sensible to align the aria selector syntax with the css selector syntax to make it easier for developers to pick up.

Reply:
Since we haven't really figured out how the recording format should be, would you be OK with leaving this as just `JSON.stringify(selector)` for now? At least we know that it is correct, and we don't have to worry about adding a dependency right now.